### ISSUE TITLE
Separate operators from functions

### ISSUE THREAD

**Issue by kareman** [realm/jazzy#233]
Even though Swift operators are technically functions I think it would be useful to at least optionally list operators under their own heading. Especially in [this case](http://kareman.github.io/SwiftShell/) where there is a function `$` which now looks like an operator.

**Comment by jpsim** [comment]
Thanks for the feature request! I don't know if it's possible for us to do, but we'll keep this open to track it anyway.

**Comment by nickplee** [comment]
+1

**Comment by p2** [comment]
Would be great to have! One of my documentations _“Other Functions”_ section currently looks like this:

```
<(_:_:)
<(_:_:)
<(_:_:)
<(_:_:)
<(_:_:)
==(_:_:)
==(_:_:)
==(_:_:)
==(_:_:)
==(_:_:)
==(_:_:)
```

**Comment by jpsim** [comment]
Just curious, how do you all think this should be handled in the generated HTML?

I totally agree it's not very useful to have a bunch of `==(_:_:)` functions.

Maybe an `Operator Functions` category that uses the declaration instead of the name? For Realm Swift, it would look like this:

```
==(lhs: ErrorType, rhs: ErrorType) -> Bool
==(lhs: ObjectSchema, rhs: ObjectSchema) -> Bool
==(lhs: Property, rhs: Property) -> Bool
==(lhs: Realm, rhs: Realm) -> Bool
==(lhs: Schema, rhs: Schema) -> Bool
==(lhs: SortDescriptor, rhs: SortDescriptor) -> Bool
```

Here's an entirely different idea of how it could be displayed:

```
ErrorType == ErrorType
ObjectSchema == ObjectSchema
Property == Property
Realm == Realm
Schema == Schema
SortDescriptor == SortDescriptor
```

Implementation aside, how would this ideally work for _you_?

**Comment by segiddins** [comment]
Another option would be to jut list each operator once, and on that operators page have each implementation

**Comment by jpsim** [comment]
ooh, I like that idea

**Comment by kareman** [comment]
Yes I think I would prefer listing each operator once too. Less clutter in the menu.

**Comment by nickplee** [comment]
Thumbs up to all of this.

**Comment by segiddins** [comment]
![screenshot 2016-06-09 20 41 53](https://cloud.githubusercontent.com/assets/1946610/15951986/abe5bd18-2e82-11e6-99b8-59d94a0f88d4.png)

**Comment by segiddins** [comment]
This is also an option:
![screenshot 2016-06-09 20 50 12](https://cloud.githubusercontent.com/assets/1946610/15952111/c9b501c2-2e83-11e6-9af8-fb310927a9f0.png)

**Comment by p2** [comment]
I'd not need the operators listed in the menu, just appear on the page of the type definition. So whenever I look at the type, I se that it conforms to `Equatable` and co.

**Comment by segiddins** [comment]
@p2 the trouble with that is that operators (and functions in general) don't 'belong' to any type

**Comment by p2** [comment]
Sure, and probably only few add an empty `extension Type: Equatable { }` before the definition. Still, it's what I'd like to look my docs like. :)

**Comment by p2** [comment]
FWIW, Xcode 8b5 now allows to move these into the type bodies, wohoo!

```swift
public struct Foo: Equatable {
    var value: String
    public static func ==(lhs: Foo, rhs: Foo) -> Bool {
        return lhs.value == rhs.value
    }
}
```

**Comment by istx25** [comment]
I believe this issue has been resolved as per @p2's last comment. Feel free to re-open if there's more to be done that I missed. Still getting familiar with the project. :)

**Comment by jpsim** [comment]
No, I think @p2's excitement was more for the language feature than what this ticket is tracking.

**Comment by istx25** [comment]
I realized that afterwards. My apologies - sounds good though.

**Comment by reitzig** [comment]
FWIW, there is some overlap with #787 here regarding how to display operators. For non-symmetric operators like, say, `<=(lhs: Int, rhs: Int/Double/...)` the default `<=(_:_:)` is still not helpful.
